BE/src/main/java/com/example/demo/Controller/CompanyController.java

```diff
@@ -0,0 +1,52 @@
+package com.example.demo.Controller;
+
+
+import com.example.demo.DTO.CompanyDTO;
+import com.example.demo.DTO.SplitDTO;
+import com.example.demo.Model.BusinessOwner;
+import com.example.demo.Model.Company;
+import com.example.demo.Persistence.API.RepositoryFactory;
+import com.example.demo.Service.BusinessOwnerService;
+import com.example.demo.Service.CompanyService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequiredArgsConstructor
+public class CompanyController {
+
+    private final CompanyService companyService;
+    private final BusinessOwnerService businessOwnerService;
+
+    @GetMapping("/all-company")
```

Comment:
REST URL design is not really the best :)  https://slides.com/spet/utcn-sd-assignment-3#/3/2